### PR TITLE
VAAPI: add vp8 support

### PR DESCRIPTION
**Pull request by lrusak**
used on Gemini lake hardware

Kodi log: http://ix.io/19fI
```
12:33:32.155 T:139942141953792   DEBUG: CDVDVideoCodecFFmpeg - Updated codec: ff-vp8-vaapi
```
vainfo relevance:
```
      VAProfileVP8Version0_3          :	VAEntrypointVLD
```

### CODE REVIEW

**Comment by FernetMenta** [comment]
does this work reliable for all vp8 files?

**Comment by lrusak** [comment]
I haven't had any issues yet, most of my vp8 files are just videos I've downloaded from youtube using youtube-dl

**Comment by FernetMenta** [comment]
The background of my question: we have settings to disable codecs supported by vaapi (all  but h.264). IMO it would make sense making vp8 configurable too.

**Comment by lrusak** [comment]
should I do it for vp9 also?

**Review by FernetMenta — APPROVED** [review]

File: xbmc/settings/Settings.h [hunk 1]
```diff
@@ -121,6 +121,8 @@ class CSettings : public CSettingsBase, public CSettingCreator, public CSettingC
   static const std::string SETTING_VIDEOPLAYER_USEVAAPIMPEG2;
   static const std::string SETTING_VIDEOPLAYER_USEVAAPIMPEG4;
   static const std::string SETTING_VIDEOPLAYER_USEVAAPIVC1;
+  static const std::string SETTING_VIDEOPLAYER_USEVAAPIVP8;
```
Comment (FernetMenta) [inline xbmc/settings/Settings.h]: btw: I don't think we should continue touching general files for platform specific behaviour. I see zero benefit in having those constants. For querying the setting in a single place it requires two entries in two differetn files.

**Comment by lrusak** [comment]
tested both settings and they are working properly on gemini lake.

**Pull request merged**